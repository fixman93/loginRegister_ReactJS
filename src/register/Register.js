import React, { Component } from 'react'

class Register extends Component {

  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }
  render() {
    return (
      <div>
        <form>
          <label htmlFor="registerEmail">
            <input type="text" id="registerEmail" placeholder="Email" onChange={event => this.setState({ email: event.target.value })} />
          </label>
          <label htmlFor="registerPassword">
            <input type="password" id="registerPassword" placeholder="Password" onChange={event => this.setState({ password: event.target.value })} />
          </label>
          <button onClick={this.handleRegistration}>Register</button>
        </form>
      </div>
    )
  }
}

export default Register