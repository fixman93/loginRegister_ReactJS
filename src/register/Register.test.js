import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import { Register } from './Register'

configure({ adapter: new Adapter() })

describe('Render Register Component', () => {
  const mockregisterUser = jest.fn()
  const props = { registerUser: mockregisterUser }
  const register = (shallow(<Register {...props} />))

  it('Register Snapshot', () => {
    expect(register).toMatchSnapshot()
  })

  describe('When typo into email input', () => {
    const email = 'test@test.com'
    beforeEach(() => {
      register.find('#registerEmail').simulate('change', { target: { value: email } })
    })

    it('Update email in `state`', () => {
      expect(register.state().email).toEqual(email)
    })
  })

  describe('When typo into password input', () => {
    const password = '123'
    beforeEach(() => {
      register.find('#registerPassword').simulate('change', { target: { value: password } })
    })

    it('Update password in `state`', () => {
      expect(register.state().password).toEqual(password)
    })
  })
})