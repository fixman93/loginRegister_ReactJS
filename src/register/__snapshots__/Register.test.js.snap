// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Register Component Register Snapshot 1`] = `ShallowWrapper {}`;
